### PR TITLE
Bug 1173106 - [Contacts][NGA] Move contacts.Matcher to it own Object.…

### PR DESCRIPTION
**Pull request by borjasalguero**
… r=arcturus

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=021342d38d78f1097da2ac7dcf04320ff6506978

File: apps/communications/contacts/js/views/multiple_select.js [hunk 1]
```diff
@@ -36,14 +38,16 @@ Contacts.MultipleSelect = (function() {
         parsedContacts = 0;
     const DEPENDENCIES = [
       '/shared/js/contacts/import/utilities/status.js',
+      '/shared/js/contacts/matcher_obj.js',
       '/shared/js/simple_phone_matcher.js',
+      '/shared/js/contacts/matcher_obj.js',
```
Comment (arcturus) [inline apps/communications/contacts/js/views/multiple_select.js]: Adding this dependency twice

File: shared/js/contacts/contacts_matcher.js [hunk 2]
```diff
@@ -1,710 +1,7 @@
...
-  Object.defineProperty(this, 'dataProvider', {
-    set: function(theProvider) {
-      dataProvider = theProvider;
-    }
-  });
-};
-
-contacts.Matcher = new contacts.MatcherObj(navigator.mozContacts);
+(function(exports) {
+  exports.Matcher = new MatcherObj(navigator.mozContacts);
```
Comment (arcturus) [inline shared/js/contacts/contacts_matcher.js]: Leaving this file requires to have imported before the 'matcher_obj.js' file, why don't keep one single file, this one, that is already imported and we don't have to add more dependencies, but this file defines the `Matcher` singleton.

I like to remove the 'kind of' namespace contact and have access to the object directly.

Comment (arcturus) [inline shared/js/contacts/contacts_matcher.js]: Also if it's necessary to expose the MatcherObject, that can be done here, a file can export more than one class.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b279dac032cc289a21bcb8961bd897b3467713eb

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2feb54aa0e93b61d0d17014e5239e5bd738bef41

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a460b6be1d625ee8bbd952672080a5cded6bc9a5

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4383121cb54f8b3c5daeb4c0fee2faf0cf9004e5

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2029a80ef4b46444408fa1666d5b64eb96f73f56

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fef7463fbf3f3184ebfcffab602b29f28060af6a

**Pull request merged**